### PR TITLE
Filter by specific loadout

### PR DESCRIPTION
**Pull request by nev-r**
image
this sets up the suggestion generator context to establish ongoing suggestion generation instead of one-time

it also teaches autocompleter to stop treating the end of a quoted phrase as still needing suggestions, to avoid this:
image

### CODE REVIEW

File: src/app/dim-api/reducer.ts [hunk 1]
```diff
@@ -980,16 +980,22 @@ function trackTriumph(
   draft.updateQueue.push(updateAction);
 }
 
-function searchUsed(draft: Draft<DimApiState>, account: DestinyAccount, query: string) {
-  const destinyVersion = account.destinyVersion;
-
-  // Real hack to fake out enough store to select out the search configs
-  const searchConfigs = searchConfigSelector(({
+// Real hack to fake out enough store to select out the search configs
+function stubSearchRootState(account: DestinyAccount) {
```
Comment (bhollis) [inline src/app/dim-api/reducer.ts:984]: Thanks

File: src/app/search/search-filters/loadouts.tsx [hunk 2]
```diff
@@ -5,9 +5,34 @@ import { FilterDefinition } from '../filter-types';
...
-      const loadoutItemIds = collectItemsInLoadouts(loadouts);
+    filter: ({ filterValue, loadouts }) => {
+      // the default search:
+      // is:inloadout
+      let selectedLoadouts = loadouts;
+
+      // a search like
+      // inloadout:"loadout name here"
+      if (filterValue !== 'inloadout') {
+        const foundLoadout = loadouts.find((l) => l.name === filterValue);
```
Comment (bhollis) [inline src/app/search/search-filters/loadouts.tsx:31]: Would you want to support a substring of the loadout name?

**Review by bhollis — APPROVED** [review]
This is great!

File: src/app/search/search-filters/loadouts.tsx (hunk 2 above)
Comment (nev-r) [reply src/app/search/search-filters/loadouts.tsx:31]: not sure i understand. if you mean while typing,
`cry` 
should recommend 
`inloadout:"deep stone crypt"`
per usual partial matching

Comment (nev-r) [reply src/app/search/search-filters/loadouts.tsx:31]: maybe you mean
`inloadout:cry`
which would be delightful but didn't quite present as low hanging fruit

Comment (bhollis) [reply src/app/search/search-filters/loadouts.tsx:31]: Yeah I meant that latter one - `inloadout:crypt` in the same vein as `perkname:clip`

Comment (nev-r) [reply src/app/search/search-filters/loadouts.tsx:31]: seems doable but will end up a bit hacky with our current low tech suggestion list. be happy to look at implementing it separately

Comment (bhollis) [reply src/app/search/search-filters/loadouts.tsx:31]: Yeah not blocking this at all. No need to generate suggestions for it either, just supporting it in the search.

**Pull request merged**